x-pack/plugins/enterprise_search/server/routes/enterprise_search/engines_preview.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { schema } from '@kbn/config-schema';
+
+import { RouteDependencies } from '../../plugin';
+
+export function registerEnginesSearchRoutes({
+  router,
+  enterpriseSearchRequestHandler,
+}: RouteDependencies) {
+  router.get(
+    {
+      path: '/internal/enterprise_search/engines/{engine_name}/search',
+      validate: {
+        body: schema.object({
```

Comment:
the request body will be an elasticsearch query DSL so wont meet these validation rules. Im also questioning the value of validation rules here, since this is a passthrough API and the ent-search API will perform the validation checks too?

The other challenge with validation rules here is updating engine APIs will require changes from two places aswell. cc @TattdCodeMonkey

Reply:
> the request body will be an elasticsearch query DSL so wont meet these validation rules

noticed some comments in [packages/core/http/core-http-server/src/router/route.ts](https://github.com/elastic/kibana/blob/a42259053749893063f522144969a0a3da35c5be/packages/core/http/core-http-server/src/router/route.ts#L173).  Assumed that query DSL could be validated if specified in the validation ?

Reply:
Ah interesting! 

We could just bypass the body validation by using `body: schema.object({}, { unknowns: 'allow' })`

Also schema config doc might have some better alternatives too: [packages/kbn-config-schema/README.md](https://github.com/elastic/kibana/blob/main/packages/kbn-config-schema/README.md)

Reply:
`body: schema.object({}, { unknowns: 'allow' })` looks a better option to bypass the validation. Also wanted to raise optional approach to use[ schema.recordOf](https://github.com/elastic/kibana/blob/main/packages/kbn-config-schema/README.md#schemarecordof)

[packages/kbn-config-schema/README.md#schemaobject](https://github.com/elastic/kibana/blob/main/packages/kbn-config-schema/README.md#schemaobject) mentions `Using unknowns: 'allow' is discouraged and should only be used in exceptional circumstances. Consider using schema.recordOf() instead.`

May be the body could be modified as: 

```
const engineSearchSchema = {
  fields: schema.recordOf(schema.string(), schema.arrayOf(schema.string())),
  query: schema.recordOf(schema.string(), schema.string()),
};
      validate: {
        body: schema.object(engineSearchSchema),
        params: schema.object({
          engine_name: schema.string(),
          from: schema.maybe(schema.number()),
          size: schema.maybe(schema.number()),
        }),

```

Reply:
One other thing to notice with this approach is that `GET` request with body would look like,

```
body: {
   query: { someQuery: 'test-sample-query' },
   fields: { someFields: ['test-name', 'test-type' ] },
},
```
Which is different from the original API body

```
  "query": {
    "multi_match": {
      "query": "someQuery",
      "fields": ["field1"]
    }
```

Reply:
May be we could circle back to use [schema.recordOf](https://github.com/elastic/kibana/blob/main/packages/kbn-config-schema/README.md#schemarecordof) later and proceed this PR with `body: schema.object({}, { unknowns: 'allow' })`

Reply:
IMO its fine to allow unknowns because ultimately ent-search API / Elasticsearch will be validating the request and provide better error messaging for incorrect use.